### ISSUE TITLE
Port the vfs sftp adapter to new async backends

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#2817]
|<img alt="cablehead's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| cablehead reported|
|-|-|
|Trac ID|trac#2817|
|Type|enhancement|
|Created|2007-09-06 10:04:38Z|

Depends on [#2815](https://github.com/twisted/twisted/issues/2815)

Check out the tests - it might be good to do something similar to [#7212](https://github.com/twisted/twisted/issues/7212) for the sftp vfs tests.

<details><summary>Searchable metadata</summary>

```
trac-id__2817 2817
type__enhancement enhancement
reporter__cablehead cablehead
priority__normal normal
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__wontfix wontfix
component__vfs vfs
keywords__ 
time__1189073078000000 1189073078000000
changetime__1300038185000000 1300038185000000
version__None None
owner__cablehead cablehead

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_1"></a>|@exarkun set status to `closed`|
|-|-|

VFS has been removed ([#4934](https://github.com/twisted/twisted/issues/4934))
